DFC.ServiceTaxonomy.Editor/Startup.cs

```diff
@@ -21,6 +23,14 @@ public void ConfigureServices(IServiceCollection services)
                 options.InstrumentationKey = Configuration["APPINSIGHTS_INSTRUMENTATIONKEY"]);
 
             services.AddOrchardCms();
+
+            services.AddEventGridPublishing(Configuration);
+
+            //todo: do this in each library??? if so, make sure it doesn't add services or config twice
+            services.AddGraphCluster(options =>
```

Comment:
We might want to do a string null check on the section to make sure we're not binding to null. If you already check later on though, ignore this comment.

Reply:
The MSDN states that `GetSection()` returns `null` if the section is missing, except it doesn't, it returns a valid (empty) section instead 😄.
So instead, I've added checks for missing config items in `GraphClusterBuilder` and it now throws a `ConfigurationErrorsException` is there are no endpoints or replica sets configured.